src/pybind/mgr/cephadm/osd.py

```diff
@@ -3,21 +3,30 @@
 import logging
 import time
 
-from typing import List, NamedTuple, Dict, Any, Set, Union
+from typing import List, Dict, Any, Set, Union
 
 import orchestrator
 from orchestrator import OrchestratorError
 
 logger = logging.getLogger(__name__)
 
 
-class OSDRemoval(NamedTuple):
```

Comment:
`NamedTuple` was problematic?

Reply:
It doesn't allow in-place updating (Unless using some [hacks](https://stackoverflow.com/a/31253184)). I update `pg_count` later. 

Seems CentOS7 ships Python 3.6. If we have at least 3.7, then [dataclasses](https://docs.python.org/3.8/library/dataclasses.html) make life much easier in this case.

Reply:
cc @jschmid1 ^^

Reply:
well.. I don't think we have 3.7 available across the board. I'd also like to have dataclasses but I think that's not possible _right_now_